1. Описать объект HttpRequest, HttpResponse, у котого есть полями Cod…

maryiadd11 opened the pull request:
1. Описать объект HttpRequest, HttpResponse, у котого есть полями Code, Headers, Body - описала обьект Request, осталось непонятный как для корректности запроса нужно делать поле headers тк hashmap по дефолту пишет через знак =; Обьект Response описан однако как записать в 2 разных файла - не понятно, из-за того что читается все посимвольно
2. Реализовать сохрание ответа на запрос в два файла (response data, response body) - описано выше, здесь результат записывается в 1 файл
3. Научить сервер обрабатывать (GET, POST, PUT DELETE) - сделано, пока без базы данных

Commit:
1. Описать объект HttpRequest, HttpResponse, у котого есть полями Code, Headers, Body -
описала обьек Request, осталось непонятный как для корректности запроса нужно делать поле headers тк hashmap по дефолту пишет через знак =
Обьект Response описан однако как записать в 2 разных файла - не понятно, из-за того что читается все посимвольно
2. Реализовать сохрание ответа на запрос в два файла (response data, response body) - описано выше, здесь результат записывается в 1 файл
3. Научить сервер обрабатывать (GET, POST, PUT DELETE) - сделано, пока без базы данных
4. Реализовать многопоточный сервер

RomanShcherbich commented on src/client/Client.java:
```diff
@@ -0,0 +1,34 @@
+        Socket client = new Socket("google.com", 80);
+        InputStream is = client.getInputStream();
+        OutputStream os = client.getOutputStream();
+        Request request1 = new Request("GET");
```
Лучше присваивать имена переменным, которые описывают их значение, например requestGet. Это нужно для более простой читаемости кода.

maryiadd11 replied:
исправлено

RomanShcherbich commented on src/entity/Request.java:
```diff
@@ -0,0 +1,70 @@
+public class Request {
+
+    private String method;
+    private final String protocol = " / HTTP/1.1";
```
по JAVA конфенции константа описывается:
`private static final PROTOCOL = "HTTP/1.1"`

maryiadd11 replied:
исправлено

RomanShcherbich commented on src/entity/Request.java:
```diff
@@ -0,0 +1,70 @@
+public class Request {
+
+    private String method;
+    private final String protocol = " / HTTP/1.1";
```
Убрать из переменной protocol URN 
" / HTTP/1.1"
/ - URN (может быть такого вида /index.html например
HTTP/1.1 - протокол

maryiadd11 replied:
поменяла

RomanShcherbich commented on src/entity/Response.java:
```diff
@@ -0,0 +1,32 @@
+package entity;
+
+public class Response {
```
Добавить в response Response code и Response description
201 - Response code
Created -  Response description

maryiadd11 replied:
+

RomanShcherbich commented on src/server/ThreadedServer.java:
```diff
@@ -0,0 +1,45 @@
+            BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(client.getOutputStream()));
+            String request = br.readLine();
+            System.out.println(String.format("Server received request [%s]",request));
+            Thread.sleep(2000);
```
Убрать слип, она только для демонстрации был

maryiadd11 replied:
+

Commit:
Исправлены недочеты.
осталось все еще не понятно: 1) как поделить запись файла на 2 разных файла; 2) как при этом использовать обьект response; 3) запрос сейчас все еще выглядит как GET / HTTP/1.1 {charset=UTF-8, Host=www.google.com, Content-Type= text/plain;} - то есть метод toString переопределен но все таки как корректно вывести hashmap осталось непонятно, гугл не помог
